### DISCUSSION TITLE
Automatic Export of Continuous S11 Measurements

### DISCUSSION THREAD

**Discussion by BioMELab-rsjensen** [jankae/LibreVNA#260] [Q&A]
First, thank you in advance for the help. Our lab collects S11 data in order to track and measure physiological states using a radio frequency resonator. So far, the Librevna has done well for us in collecting single sweeps then exporting them for analysis. However, we'd like to be able to collect data from continuous sweeps over time.

Is there a method for automatically exporting each sweep as the device sweeps continuously? If not, what suggestions do you have for us?

**Comment by jankae** [comment]
> Is there a method for automatically exporting each sweep as the device sweeps continuously?

Yes there is. But I only implemented it a couple of days ago, that likely explains why you haven't found it yet. You will need the latest development build (see [here ](https://github.com/jankae/LibreVNA/blob/master/Documentation/DeveloperInfo/DownloadActionBuilds.md)on how to download it).
Please refer to the [programming guide](https://github.com/jankae/LibreVNA/blob/master/Documentation/UserManual/ProgrammingGuide.pdf) (chapter 5: Streaming data) on how to use this new feature. There is also an [example python script](https://github.com/jankae/LibreVNA/blob/master/Documentation/UserManual/SCPI_Examples/capture_live_data.py) demonstrating how to stream VNA data.

**Reply by BioMELab-rsjensen** [reply]
Thank you once again for your help. I have to admit I'm a novice when it comes to this sort of thing, so please bear with me.

Using the latest (1.6.0) GUI, Capture_live_data.py, and libreVNA.py elements, I've run into the following error:

`Traceback (most recent call last):
  File "c:\Users\reill\OneDrive - Wichita State University\Reilly Jensen - Cluff Lab\VNWA\LibreVNA\Scripts\capture_live_data.py", line 13, in <module>
    vna.cmd(":DEV:CONN")
  File "c:\Users\reill\OneDrive - Wichita State University\Reilly Jensen - Cluff Lab\VNWA\LibreVNA\Scripts\libreVNA.py", line 98, in cmd
    status = self.get_status(timeout=timeout)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\reill\OneDrive - Wichita State University\Reilly Jensen - Cluff Lab\VNWA\LibreVNA\Scripts\libreVNA.py", line 119, in get_status
    raise Exception("Expected numeric response from *ESR? but got "
Exception: Expected numeric response from *ESR? but got '{"Z0":50.0,"dBm":-10.0,"frequency":3132478000.0,"measurements":{"S11_imag":0.3370531719163283,"S11_real":0.33979116630937145,"S12_imag":-3.456435713519591e-05,"S12_real":3.906903119689978e-05,"S21_imag":4.816743711384793e-05,"S21_real":9.816035757770978e-06,"S22_imag":0.26530173394403106,"S22_real":-0.3552225403222953},"pointNum":261}'
PS C:\Users\reill\OneDrive - Wichita State University\Reilly Jensen - Cluff Lab\VNWA\LibreVNA\Scripts>`

Any help would be greatly appreciated. If I manage to figure out the issue before your response, I will reply accordingly. Again, thank you for your help!

**Reply by jankae** [reply]
Did you modify the python script in any way? And which ports do you have configured in the preferences of the GUI?
Your error sounds like the SCPI client (the python script) is trying to connect to the live data stream instead of the SCPI server.

**Reply by jankae** [reply]
Additional clarification, because I think it is easy to get that part wrong:
The `Capture_live_data.py` is talking to both the SCPI server (to connect the device, set up and start the sweep) and the server for live raw data (to actually capture the data). The SCPI server is expected to be at port 19542, the live raw data server at port 19000. That means you need to configure the preferences like this:
![image](https://github.com/user-attachments/assets/2065e092-1300-41b3-9450-1969bb72ec60)
![image](https://github.com/user-attachments/assets/5e887db3-5e22-45ec-8b05-84b3c032e402)
Based on your error message, I am guessing that you have enabled the raw data streaming server at port 19542 instead.

**Reply by BioMELab-rsjensen** [reply]
Yes, that seems to exactly be the issue! Thank you for the help. I figured it was something simple that I was overlooking.
